src/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs

```diff
@@ -127,234 +121,533 @@ private DependencyContext Read(JObject root)
             }
 
             return new DependencyContext(
-                new TargetInfo(target, runtime, runtimeSignature, isPortable),
-                ReadCompilationOptions((JObject)root[DependencyContextStrings.CompilationOptionsPropertName]),
-                ReadLibraries(compileTarget, false, libraryStubs).Cast<CompilationLibrary>().ToArray(),
-                ReadLibraries(runtimeTarget, true, libraryStubs).Cast<RuntimeLibrary>().ToArray(),
-                ReadRuntimeGraph((JObject)root[DependencyContextStrings.RuntimesPropertyName]).ToArray()
-                );
+                new TargetInfo(framework, runtime, runtimeSignature, isPortable),
+                compilationOptions,
+                CreateLibraries(compileTarget?.Libraries, false, libraryStubs).Cast<CompilationLibrary>().ToArray(),
+                CreateLibraries(runtimeTarget.Libraries, true, libraryStubs).Cast<RuntimeLibrary>().ToArray(),
+                runtimeFallbacks ?? Enumerable.Empty<RuntimeFallbacks>());
         }
 
-        private IEnumerable<RuntimeFallbacks> ReadRuntimeGraph(JObject runtimes)
+        private Target SelectRuntimeTarget(List<Target> targets, string runtimeTargetName)
         {
-            if (runtimes == null)
+            Target target;
+
+            if (targets == null || targets.Count == 0)
+            {
+                throw new FormatException("Dependency file does not have 'targets' section");
+            }
+
+            if (!string.IsNullOrEmpty(runtimeTargetName))
+            {
+                target = targets.FirstOrDefault(t => t.Name == runtimeTargetName);
+                if (target == null)
+                {
+                    throw new FormatException($"Target with name {runtimeTargetName} not found");
+                }
+            }
+            else
             {
-                yield break;
+                target = targets.FirstOrDefault(t => IsRuntimeTarget(t.Name));
             }
 
-            foreach (var pair in runtimes)
+            return target;
+        }
+
+        private bool IsRuntimeTarget(string name)
+        {
+            return name.Contains(DependencyContextStrings.VersionSeperator);
+        }
+
+        private void ReadRuntimeTarget(JsonTextReader reader, out string runtimeTargetName, out string runtimeSignature)
+        {
+            runtimeTargetName = null;
+            runtimeSignature = null;
+
+            reader.Read();
+            if (reader.TokenType == JsonToken.String)
             {
-                yield return new RuntimeFallbacks(pair.Key, pair.Value.Values<string>().ToArray());
+                // This fallback is temporary
```

Comment:
How temporary is this fallback?  Should this comment be removed?  Should this check?